src/store/modules/DownloadModule.ts

```diff
@@ -125,28 +127,67 @@ export const DownloadModule = {
             });
         },
 
-        async downloadAndInstallSpecific({state, commit, dispatch}, params: {
-            combo: ThunderstoreCombo,
-            profile: ImmutableProfile
+        async download({state, commit, dispatch}, params: {
+            combos: ThunderstoreCombo[],
+            downloadId: UUID
         }) {
-            const downloadId = await dispatch('addDownload', [`${params.combo.getMod().getName()} (${params.combo.getVersion().getVersionNumber().toString()})`]);
-
             try {
-                const downloadedMods = await ThunderstoreDownloaderProvider.instance.download(
-                    params.profile,
-                    params.combo,
+                await ThunderstoreDownloaderProvider.instance.downloadCombos(
+                    params.combos,
                     state.ignoreCache,
-                    (downloadProgress: number, modName: string, status: number, err: R2Error | null) => {
-                        dispatch('downloadProgressCallback', { downloadId, downloadProgress, modName, status, err });
+                    (downloadProgress, modName, status, err) => {
+                        dispatch('downloadProgressCallback', { downloadId: params.downloadId, downloadProgress, modName, status, err });
                     }
                 );
-                await dispatch('installMods', {downloadedMods, downloadId, profile: params.profile});
             } catch (e) {
-                commit('updateDownload', { downloadId, failed: true });
-                throw e;
+                commit('updateDownload', { downloadId: params.downloadId, failed: true });
+                commit('error/handleError', R2Error.fromThrownValue(e), { root: true });
```

Comment:
Why handle the error hear instead of throwing it? Now it looks like `downloadAndInstallCombos` doesn't know if this call fails and will continue to the install step even if the mods weren't downloaded.

Reply:
I changed it back to `throw`.